Frontend

markusyeo opened the pull request:
- The page shows a loading page while checking the status of the page
- Create a new user profile page
- Change edit page to modal
- Add user update details functionality
Backend:
- Update user route to allow updating of preferredComplexity and preferredLanguage

markusyeo committed:
Wrap main app in a full width box

markusyeo committed:
Update user page

- Now we have a user page instead of defaulting to the edit view
- Edit user view is now a modal instead

markusyeo committed:
Update backend userRouter

- Allow updating of user preferred complexity and language
- Remove updating of email and id

markusyeo committed:
Update EditUserModal

- EditUserModal now takes variables from UserProfilePage rather than calling the backend for it's own copy of user details

markusyeo committed:
Neaten up the homepage

- Honestly we might not need a homepage

markusyeo committed:
Fix a bug that sends the wrong updated username

markusyeo committed:
Wrap app in cssbaseline

- Tidy up other redundant dependencies and errors

markusyeo committed:
Update app.tsx

- Now there is a loading page before the session is retrieved from supabase

markusyeo committed:
Tidy up some console logs

markusyeo committed:
Clean up userpage

- User can only edit if the profile is their own

ryanpeh commented on frontend/src/components/User/UserProfileContainer.tsx:
```diff
@@ -0,0 +1,97 @@
+    isError
+  } = userData;
+
+  if (isLoading) {
+    return <Loading />;
+  }
```
Remove this

ryanpeh commented on frontend/src/components/User/UserProfileContainer.tsx:
```diff
@@ -0,0 +1,97 @@
+    isError
+  } = userData;
+
+  if (isLoading) {
+    return <Loading />;
+  }
+
+  return isError ? (
+    <Loading />
```
Change to error page

ryanpeh commented on frontend/src/components/User/UserProfileContainer.tsx:
```diff
@@ -0,0 +1,97 @@
+    userName,
+    userPreferredComplexity,
+    userPreferredLanguage,
+    isLoading,
```
Can remove this if not needed anymore

markusyeo committed:
fix: wrong truthfulness for editable user

- Remove loading page on user container

markusyeo committed:
chore: move NotFound error page to components

markusyeo committed:
fix: Restore the loading and error pages in user container

- The hook in user container is used to retrieve the user id for the provided id
